website/content/docs/platform/mssql/installation.mdx

```diff
@@ -0,0 +1,190 @@
+* An authenticated Vault client
+
+[linux-ekm]: https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-editions-and-components-2019?view=sql-server-ver15#Unsupported
+
+## Installing the Vault EKM provider
+
+## Configuring Vault
+
+The EKM provider requires AppRole auth and the Transit secret engine to be setup
+on the Vault server. The steps below can be used to configure Vault ready for the
+EKM provider to use it.
+
+-> **Note:** The first release of the Vault EKM provider does not support custom
+mount paths or namespaces for the AppRole and Transit engines.
+
+-> **Note:** rsa-2048 is currently the only supported key type.
+
+1. Set up AppRole auth:
+
+    ```bash
+    vault auth enable approle
+    vault write auth/approle/role/tde-role \
+        token_ttl=20m \
+        max_token_ttl=30m \
+        token_policies=tde-policy
+    ```
+
+1. Retrieve the AppRole ID and secret ID for use later when configuring SQL Server:
+
+    ```bash
+    vault read auth/approle/role/ekm-encryption-key-role/role-id
+    vault write -f auth/approle/role/ekm-encryption-key-role/secret-id
+    ```
+
+1. Enable the transit secret engine and create a key:
+
+    ```bash
+    vault secrets enable transit
+    vault write -f transit/keys/ekm-encryption-key type="rsa-2048"
+    ```
+
+1. Create a policy for the Vault EKM provider to use. The following policy has
+    the minimum required permissions:
+
+    ```bash
+    vault policy write tde-policy -<<EOF
+    path "transit/keys/ekm-encryption-key" {
+        capabilities = ["create", "read", "update", "delete"]
+    }
+
+    path "transit/keys" {
+        capabilities = ["list"]
+    }
+
+    path "transit/encrypt/ekm-encryption-key" {
+        capabilities = ["update"]
+    }
+
+    path "transit/decrypt/ekm-encryption-key" {
+        capabilities = ["update"]
+    }
+
+    path "sys/license/status" {
+        capabilities = ["read"]
+    }
+    EOF
+    ```
+
+## Configuring SQL Server
+
+The remaining steps are all run on the database server.
+
+### Install the EKM provider on the server
+
+1. Download and run the latest Vault EKM provider installer from
+  [releases.hashicorp.com](https://releases.hashicorp.com/vault-mssql-ekm-provider/)
+
+1. Enter your Vault server's address when prompted and complete the installer
+
+### Configure the EKM provider using SQL
+
+1. Enable the EKM feature and create a cryptographic provider using the folder
+   you just installed the EKM provider into.
+
+    ```sql
+    -- Enable advanced options
+    USE master;
+    GO
+
+    EXEC sp_configure 'show advanced options', 1;
+    GO
+
+    RECONFIGURE;
+    GO
+
+    -- Enable EKM provider
+    EXEC sp_configure 'EKM provider enabled', 1;
+    GO
+
+    RECONFIGURE;
+    GO
+
+    CREATE CRYPTOGRAPHIC PROVIDER TransitVaultProvider
+    FROM FILE = 'C:\Program Files\HashiCorp\Transit Vault EKM Provider\TransitVaultEKM.dll'
+    GO
+    ```
+
+1. Next, create credentials for an admin to use EKM with your AppRole role and
```

Comment:
Does the `with IDENTITY` parameter correspond to the AppRole role id? I wonder if there's an easy way to make it a little more clear what the parameters in the sql snippet correlate to in vault, like with comments or something.

Reply:
Yeah good point, I've made it a bit clearer now.